pkg/oc/cli/cmd/image/append/append.go

```diff
@@ -0,0 +1,635 @@
+		Add layers to Docker images
+
+		Modifies an existing image by adding layers or changing configuration and then pushes that
+		image to a remote registry. Any inherited layers are streamed from registry to registry 
+		without being stored locally. The default docker credentials are used for authenticating 
+		to the registries.
+
+		Layers may be provided as arguments to the command and must each be a gzipped tar archive
+		representing a filesystem overlay to the inherited images. The archive may contain a "whiteout"
+		file (the prefix '.wh.' and the filename) which will hide files in the lower layers. All
+		supported filesystem attributes present in the archive will be used as is.
+
+		Metadata about the image (the configuration passed to the container runtime) may be altered
+		by passing a JSON string to the --image or --meta options. The --image flag changes what
+		the container runtime sees, while the --meta option allows you to change the attributes of
+		the image used by the runtime. Use --dry-run to see the result of your changes. You may
+		add the --drop-history flag to remove information from the image about the system that 
+		built the base image.
+
+		Images in manifest list format will automatically select an image that matches the current
+		operating system and architecture unless you use --filter-by-os to select a different image.
+		This flag has no effect on regular images.
+
+		Experimental: This command is under active development and may change without notice.`)
+
+	example = templates.Examples(`
+# Remove the entrypoint on the mysql:latest image
+%[1]s --from mysql:latest --to myregistry.com/myimage:latest --image {"Entrypoint":null}
+
+# Add a new layer to the image
+%[1]s --from mysql:latest --to myregistry.com/myimage:latest layer.tar.gz
+`)
+)
+
+type options struct {
+	Out, ErrOut io.Writer
+
+	From, To   string
+	LayerFiles []string
+
+	ConfigPatch string
+	MetaPatch   string
+
+	DropHistory bool
+
+	OSFilter        *regexp.Regexp
+	DefaultOSFilter bool
+
+	FilterByOS string
+
+	MaxPerRegistry int
+
+	DryRun   bool
+	Insecure bool
+	Force    bool
+}
+
+// schema2ManifestOnly specifically requests a manifest list first
+var schema2ManifestOnly = distribution.WithManifestMediaTypes([]string{
+	manifestlist.MediaTypeManifestList,
+	schema2.MediaTypeManifest,
+})
+
+// New creates a new command
+func New(name string, out, errOut io.Writer) *cobra.Command {
+	o := &options{
+		MaxPerRegistry: 3,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "append",
+		Short:   "Add layers to images and push them to a registry",
+		Long:    desc,
+		Example: fmt.Sprintf(example, name),
+		Run: func(c *cobra.Command, args []string) {
+			o.Out = out
+			o.ErrOut = errOut
+			kcmdutil.CheckErr(o.Complete(c, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	flag := cmd.Flags()
+	flag.BoolVar(&o.DryRun, "dry-run", o.DryRun, "Print the actions that would be taken and exit without writing to the destinations.")
```

Comment:
should be destination (singular)?  Doesn't look like you're allowing multiple destinations?